pkg/sql/types/types.proto

```diff
@@ -513,4 +513,10 @@ message InternalType {
     // StableTypeID is set only for User Defined Types.
     optional uint32 stable_type_id = 15
```

Comment:
wondering if it's worth keeping this in a struct, e.g. `optional UserDefinedTypeMetadata udt_metadata...` and make this field and the one below non nullable. then it's easy to tell whether it's a UDT or not (udt_metadata != nil)

Reply:
Oh, that might be a good idea! IDK if we are worried about extra allocations when reading types protos though (maybe thats nothing to worry about).

Reply:
I'll wait until more fields pop up -- i think 3 is a good tipping point for its own struct.

Reply:
i think 2 is :3

Reply:
Done, PTAL!